### ISSUE TITLE
feat(cdk/a11y): make `_computeAriaAccessibleName` work for more general use cases

### ISSUE THREAD

**Issue by zarend** [angular/components#26104]
### Feature Description

#25667 implemented a `_computeAriaAccessibleName` method so that we can add aria labels to elements that are rendered by the component and the developer does not have access to. It only supports datepicker's use case, and allows us to add an accessible name without creating unnecessary public API.

Consider making it working more broadly so it can be used in other cases.

**Comment by crisbeto** [comment]
Closing since we haven't had a need for this so far.
